### DISCUSSION TITLE
Question about how to make BooleanVectorOutput

### DISCUSSION THREAD

**Discussion by kmerbcn** [OpenModelica/OpenModelica#12609] [General]
Hello

I am trying to figure out how to make a BooleanVectorOutputConnector in the same way as in the library.
I feel it lacks some explanation.

I wrote in modelica

connector BooleanVectorOutput = output Bolean
and it gives me an error. How can it be?

Br,

javier martinez

**Comment by perost** [comment]
You've misspelled Boolean. If that's not the actual issue then please give more details such as what error message you're getting.

**Comment by kmerbcn** [comment]
connector BooleanVectorOuput "Boolean input connector used for vector of connectors"
  annotation(
    defaultComponentName = "u",
    Icon(graphics = {Ellipse(lineColor = {255, 0, 255}, fillColor = {255, 0, 255}, lineThickness = 0.5, extent = {{-100, -100}, {100, 100}})}, coordinateSystem(extent = {{-100, -100}, {100, 100}}, preserveAspectRatio = false, initialScale = 0.2)),
    Diagram(coordinateSystem(preserveAspectRatio = false, initialScale = 0.2, extent = {{-100, -100}, {100, 100}}), graphics = {Text(textColor = {255, 0, 255}, extent = {{-10, 85}, {-10, 60}}, textString = "%name"), Ellipse(lineColor = {255, 0, 255}, fillColor = {255, 0, 255}, lineThickness = 0.5, extent = {{-50, 50}, {50, -50}})}),
    Documentation(info = "<html>
<p>
Boolean input connector that is used for a vector of connectors,
for example <a href=\"modelica://Modelica.Blocks.Interfaces.PartialBooleanMISO\">PartialBooleanMISO</a>,
and has therefore a different icon as BooleanInput connector.
</p>
</html>"));
end BooleanVectorOuput;

The model looks like that. It seems it does not give an error. My idea is to have 2 vectors of boolean output. Every vector should contain 4 boolean types inside. How can I instantiate this in a bigger model? Do you have an example?

**Reply by perost** [reply]
I'm not sure what you're really trying to do, but that connector is missing the `= output Boolean` part as it's declared in the MSL. The correct syntax would be `connector BooleanVectorOutput = output Boolean;`.

**Comment by kmerbcn** [comment]
I tried same solution by looking at the modelica definitions. Nevertheless, it gives an error to me. Strange.
I would like to create a vector of boolean in a modelica model containing 4 elements. How can I do it by using connector solutions?

Br

Javier

**Reply by kmerbcn** [reply]
Hello Per and thanks for the help!

Basically, I want to do the following model, which also gives an error to me

```modelica
 model ClassicalDAB
    parameter Real Ts;
    inner Boolean[4] H1;
    
    block State1
      outer output Boolean[4] H1;
    equation
      H1[1] = true;
      H1[2] = true;
      H1[3] = false;  
      H1[4] = false;
    end State1;
    
    block State2
      outer output Boolean[4] H1;
     equation
      H1[1] = false;
      H1[2] = true;
      H1[3] = false;
      H1[4] = false;
    end State2;
  
    State1 state1;  
    State2 state2;
    
  equation
    initialState(state1);
    transition(state1, state2, sample(0, Ts), immediate=false);
    transition(state2, state1, sample(0, Ts), immediate=false);
end ClassicalDAB;
```

Basically the error is that I have more equations than variables. How can i overcome it? By the way, is the statement sample(0, Ts) inside of transition correct?

Br,

Javier Mrtinez
